lib/src/features/metrics/metrics_controller.dart

```diff
@@ -0,0 +1,308 @@
+import 'package:fit_and_healthy/shared/models/activity_level.dart';
+import 'package:fit_and_healthy/shared/models/gender.dart';
+import 'package:fit_and_healthy/shared/models/weight_goal.dart';
+import 'package:fit_and_healthy/src/features/auth/auth_repository/firebase_auth_repository.dart';
+import 'package:fit_and_healthy/shared/models/WeightEntry.dart';
+import 'package:fit_and_healthy/src/features/user/user_model.dart';
+import 'package:fit_and_healthy/src/features/user/user_repository.dart';
+import 'package:fit_and_healthy/src/features/user/weight_repository.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'metrics_controller.g.dart';
+
+@riverpod
+class MetricsController extends _$MetricsController {
+  late final UserRepository _userRepository;
+  late final WeightRepository _weightRepository;
+
+  @override
+  Future<Map<String, dynamic>> build() async {
+    final authRepository = ref.read(firebaseAuthRepositoryProvider);
+    _userRepository = UserRepository(authRepository);
+    _weightRepository = WeightRepository(authRepository);
+
+    final userData = await _userRepository.getUser();
+    final weightHistory = await _weightRepository.getWeightHistory();
+
+    if (userData == null) throw Exception('UserModel not found');
+
+    return {
```

Comment:
Im not a fan of storing the data as a map here as it makes type safety hard to accomplish. We should be sure of the data that comes from the database already and should therefore store it in some form of entity class instead (MetricsState?)